pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -138,11 +138,13 @@ func resolveResultRef(pipelineState PipelineRunState, resultRef *v1beta1.ResultR
 		return nil, resultRef.PipelineTask, fmt.Errorf("task %q referenced by result was not successful", referencedPipelineTask.PipelineTask.Name)
 	}
 
-	var runName, taskRunName, resultValue string
+	var runName, runValue, taskRunName string
+	var resultValue v1beta1.ArrayOrString
 	var err error
 	if referencedPipelineTask.IsCustomTask() {
 		runName = referencedPipelineTask.Run.Name
-		resultValue, err = findRunResultForParam(referencedPipelineTask.Run, resultRef)
+		runValue, err = findRunResultForParam(referencedPipelineTask.Run, resultRef)
+		resultValue = *v1beta1.NewArrayOrString(runValue)
```

Comment:
should this be calling unmarshalling here? Calling NewArrayOrString with `runValue` here returns a `string`. 

Also, why limiting this to just the custom task?

Reply:
> Also, why limiting this to just the custom task?

So custom task results are still `string`:

```
type RunResult struct {
	// Name the given name
	Name string `json:"name"`
	// Value the given value of the result
	Value string `json:"value"`
}
```

Are we planning on changing the custom task result to string/array? Something coming up ...

Reply:
Yes, this is a great question! We don't support structured results for custom task in this PR. So this is intentionally to return string here.

Reply:
We plan to add support for task and pipeline first, custom task is not discussed and also not explicitly mentioned in the tep (in tep we only mention task and pipeline) but we will discuss later!

https://github.com/tektoncd/community/blob/main/teps/0076-array-result-types.md#summary

Reply:
The first use case explicitly talks about custom task - https://github.com/tektoncd/community/blob/main/teps/0076-array-result-types.md#use-cases Custom task `params` has use case to consume an array which could be a result of other task and produce a result with an array (a list of hashes/a list of built images).

Thank you @Yongxuanzhang for following up on this 🙏

Reply:
Yes! I have added in our tracking issue